### ISSUE TITLE
Issues with multiple files sharing the same name

### ISSUE THREAD

**Issue by loikki** [michaeljones/breathe#703]
Hi,

Breathe cannot differentiate two files sharing the same name (but not directory) and as trouble with structures defined multiple times.

I am currently trying to use breathe in the following C project https://gitlab.cosma.dur.ac.uk/swift/swiftsim (branch `link_doxy`). In the code, we have a system of modules in order to switch between different schemes. The idea is to have a file that includes the correct directory depending on some configuration options. For example, we can switch between `src/hydro/Gadget2` and `src/hydro/SPHENIX` that are two different implementations of hydrodynamics. It means that within the directories, we have exactly the same file, function and structure names defined (but with different implementation).

Could it be possible to have an option in breathe-apidoc / breathe to use the filenames including the directory and differentiate two implementations of the same structure? I would be happy to have all the different modules documented at the same time, no need to try to read some of our configuration options.

Example of the issue for the files:
![image](https://user-images.githubusercontent.com/14158679/119990704-f2096b00-bfc8-11eb-8981-25d87c6d08c8.png)

Example of the issue for the structures (attributes not shown):
![image](https://user-images.githubusercontent.com/14158679/119990828-182f0b00-bfc9-11eb-8d40-317e0f8aeaa0.png)

If you wish to reproduce my steps, I have done the following:
```bash
./configure
cd doc/RTD
make doxygen
make html
firefox build/html/index.html
```
I am calling breathe-apidoc like this within `make doxygen`: `breathe-apidoc -f -g struct,file ../xml/ -o "$(SOURCEDIR)/Doxygen/" -p SWIFT`.

Thank you very much for your help and time,
Loikki

**Comment by vermeeren** [comment]
@loikki I think (but have not verified) that in case of ambiguity the `doxygenfile` directive simply needs a more detailed path specification, such as `dir_a/cooling_rates.h`. If I understand correctly, the problem happens because you use `breathe-apidoc` and the intermediate output of apidoc does not properly specify the containing directory (as there is no control over this). Could you confirm this is the case?

I think it shouldn't be too hard to get a more properly specified path specification for apidoc, which should resolve the issue. I tried checking your project for reproduction, but presumably because so many time has passed things are a bit different over there compared to when you opened the issue.

**Comment by loikki** [comment]
Yes exactly, I am using `breathe-apidoc`. From what I understood of the issue, you are perfectly correct. I think simply specifying the full path should solve the issue.

The branch should still be fine. If it helps you, I can do the checks myself once the fix is implemented.

Thank you very much for your help.
